### ISSUE TITLE
Add aoflagger version check 

### ISSUE THREAD

**Issue by jradcliffe5** [e-merlin/CASA_eMERLIN_pipeline#24]
see #23 comments

**Comment by jmoldon** [comment]
Yes, I agree. We need a switch to choose between the direct ms2mms and ms2mms_fields depending on the flagger version, I will add that step and also adapt run_aoflagger accordingly. I will do it in this branch before merging it.

Because aoflagger does not run in parallel with mms, maybe ms2mms_fields will still be faster in terms of memory requirements (being able to use memory read instead of direct read in some cases) as mentioned before. But if the speed is comparable we should use the direct ms2mms for aoflagger 2.9+ and SERBob when available.

**Comment by jradcliffe5** [comment]
Great! True, I think it depends on computer architecture. I am gifted here, as I have a computer with 800GB ram.

Maybe we could add another switch (far in the future) which checks the RAM and then picks ms2mms_fields if it cannot use memory read?

I could test the timings of the aoflagger 2.7 with ms2mms_fields and aoflagger 2.9 on my machine here as I have both versions installed? I can also time and test what happens with direct and memory read.

**Comment by jmoldon** [comment]
Good thanks! If you have "too much memory", more than the rest of us in the real world, you can limit how much memory is accessible for a terminal with `ulimit -v <how-many-ks>`. I'm not sure how well can it limit the memory for casa in multi-processing mode.
Maybe you can check something like 64GB, 128GB and unlimited?

**Comment by jradcliffe5** [comment]
Sure thanks. I can also restrict aoflagger using the `-mem` argument I think. I'll make some plots of time vs. memory with the 2.7 & 2.9 versions (i.e. forced mms shape compared to optimised). I guess I can do the same for no. of cores with different available RAM vs time. 

I guess in the end we would want to test this on a 24hr run i.e. a ~300GB data set. I don't have that time on this computer atm but I will do the tests on a 106GB data set I have here.

**Comment by jradcliffe5** [comment]
Going to close with with the merge of #23, will make new issue with regards to my previous comment.
